app/routes/campaign.js

```diff
@@ -17,8 +17,11 @@ module.exports = function (router) {
     router.post('/campaign/create', (request, response) => {
         if(request.user) {
             campaignPromise = campaignController.saveCampaign(request);
-            campaignPromise.then(function(campaign) {
-                response.send(campaign._id);
+            campaignPromise.then(function() {
+                return campaignController.findCampaignByTitleAndUser(request.body.title, request.user.id);
+            })
+            .then(function(result) {
+                response.send(result[0]._id);
```

Comment:
Was this deliberate? Doesn't line 21 return in the two lines being removed successfully send the campaign id?